### PR TITLE
Refactor e3sm test suite definition format

### LINKED ISSUE #2409: Syntax of tests.py is not user friendly

### PR DESCRIPTION
**Pull request by jgfouca**
The use of tuples is not scalable as we increase the
configurability of test suites.

Fixes #2409 

[BFB]

### CODE REVIEW

**Comment by jgfouca** [comment]
This is worth a thorough review since it will impact how E3SM developers interact with CIME when defining test suites.

I have felt for a while that the syntax of our test-suites was very awkward. With the new "shared-build" field coming, with the old format, yet another None would have to be placed in all our test suites, so I figured it was time to bite the bullet and change the format. key:value should be clearer and more-scalable than position-based fields.

**Comment by jonbob** [comment]
Should I wait on 2749 then until you get this in?

**Comment by jgfouca** [comment]
@jonbob , you don't need to wait. It will be an easy conflict to resolve.

**Comment by jonbob** [comment]
thanks @jgfouca

**Comment by jgfouca** [comment]
@rljacob , want to weigh-in here?

File: cime/config/e3sm/tests.py [hunk 1]
```diff
@@ -1,136 +1,172 @@
...
-# <test>.<grid>.<compset>.
-# suite_name -> (inherits_from, timelimit, [test [, mods[, machines]]])
-#   To elaborate, if no mods are needed, a string representing the testname is all that is needed.
-#   If testmods are needed, a 2-ple must be provided  (test, mods)
-#   If you want to restrict the test mods to certain machines, than a 3-ple is needed (test, mods, [machines])
+# <test>.<grid>.<compset>[.<testmod>]
+#
+# suite_name : {
+#     "inherit" : (suite1, suite2, ...), # Optional. Suites to inherit tests from. Default is None. Tuple, list, or str.
+#     "time"    : "HH:MM:SS",            # Optional. Recommended upper-limit on test time.
```
Comment (rljacob) [inline cime/config/e3sm/tests.py]: Is that upper limit per test or for the whole group?

Comment (jgfouca) [reply cime/config/e3sm/tests.py]: @rljacob , it applies as a first estimate for all the tests in the suite/group.

**Comment by jgfouca** [comment]
Merged to next. Will need to keep a close eye on dashboard tmrw.

File: cime/config/e3sm/tests.py [hunk 2]
```diff
@@ -1,136 +1,172 @@
...
+            "NCK.ne11_oQU240.A_WCYCL1850",
+            "PET.f19_g16.X.allactive-mach-pet",
+            "PET.f45_g37_rx1.A.allactive-mach-pet",
+            "PET_Ln9_PS.ne30_oECv3_ICG.A_WCYCL1850S.allactive-mach-pet",
+            "PEM_Ln9.ne30_oECv3_ICG.A_WCYCL1850S",
+            "ERP_Ld3.ne30_oECv3_ICG.A_WCYCL1850S",
+            "SMS.f09_g16_a.MALI",
+            "SMS_D_Ln5.conusx4v1_conusx4v1.FC5AV1C-L",
+            "SMS.ne30_oECv3.BGCEXP_BCRC_CNPECACNT_1850.clm-bgcexp",
+            "SMS.ne30_oECv3.BGCEXP_BCRC_CNPRDCTC_1850.clm-bgcexp",
```
[the remark this answers is not in the document]
Comment (jgfouca) [reply cime/config/e3sm/tests.py]: @rljacob , with the current implementation, yes.

**Pull request merged**